```
>   variance = np.dot(e, e) / (ny - nb)
       > /private/tmp/nix-build-python3.12-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1471: RuntimeWarning: invalid value encountered in multiply
       >   pre_var = variance * np.dot(
       > .................
       > ======================================================================
       > ERROR: test_float128 (tests.tests.TestEverything.test_float128)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-python3.12-pwlf-2.5.0.drv-0/source/tests/tests.py", line 583, in test_float128
       >     np.linspace(0, 10, 3, dtype=np.float128),
       >                                 ^^^^^^^^^^^
       >   File "/nix/store/1qdjbsbqq9vlbn9g0amydwi4ldmwkn9m-python3.12-numpy-2.2.2/lib/python3.12/site-packages/numpy/__init__.py", line 414, in __getattr__
       >     raise AttributeError("module {!r} has no attribute "
       > AttributeError: module 'numpy' has no attribute 'float128'. Did you mean: 'float16'?
       >
       > ----------------------------------------------------------------------
       > Ran 62 tests in 12.332s
       >
       > FAILED (errors=1)
       For full logs, run 'nix log /nix/store/lpsk8my7sa2crl0zaz16d3f9p200cnh5-python3.12-pwlf-2.5.0.drv'.
error: builder for '/nix/store/n8r8z43xn27gzzsyyhs71b7i8xsdk7rk-python3.13-pwlf-2.5.0.drv' failed with exit code 1;
       last 25 log lines:
       >                        ...
       >                        [ 2.286e-01  7.723e-01]
       >                        [ 2.288e-01  7.703e-01]]
       >  population_energies: [ 8.873e-01  8.894e-01 ...  8.884e-01  8.884e-01]
       > ....../private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1461: RuntimeWarning: divide by zero encountered in scalar divide
       >   variance = np.dot(e, e) / (ny - nb)
       > /private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/pwlf/pwlf.py:1471: RuntimeWarning: invalid value encountered in multiply
       >   pre_var = variance * np.dot(
       > .................
       > ======================================================================
       > ERROR: test_float128 (tests.tests.TestEverything.test_float128)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-python3.13-pwlf-2.5.0.drv-0/source/tests/tests.py", line 583, in test_float128
       >     np.linspace(0, 10, 3, dtype=np.float128),
       >                                 ^^^^^^^^^^^
       >   File "/nix/store/km2m3g1ld3ifc3rzmsb2sbimn7qgwq8h-python3.13-numpy-2.2.2/lib/python3.13/site-packages/numpy/__init__.py", line 414, in __getattr__
       >     raise AttributeError("module {!r} has no attribute "
       >                          "{!r}".format(__name__, attr))
       > AttributeError: module 'numpy' has no attribute 'float128'. Did you mean: 'float16'?
       >
       > ----------------------------------------------------------------------
       > Ran 62 tests in 12.737s
       >
       > FAILED (errors=1)
```

--- a/pkgs/development/python-modules/pwlf/default.nix
+++ b/pkgs/development/python-modules/pwlf/default.nix
@@ -12,14 +12,14 @@
 
 buildPythonPackage rec {
   pname = "pwlf";
-  version = "2.5.0";
+  version = "2.5.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "cjekel";
     repo = "piecewise_linear_fit_py";
     tag = "v${version}";
-    hash = "sha256-3eeVFXmdbuAYVTl3iK58LXfe+ZOx+mt5GzWLxsvTfEI=";
+    hash = "sha256-tUJSVMhnUdqpbiKzrpznCEgaGHGOLylFm6DzNTnvnQQ=";
   };
 
   nativeBuildInputs = [

